File: busquedaBinaria/binaria.py
def busquedaBinaria(num):
    cont = 0
    l = 0
    r = len(num) -1
    
    while l <= r:
        m = int((l+r)/2)
        cont +=1
        
        if num[m] < numABuscar:
            l = m+1
        elif num[m] > numABuscar:
            r = m-1
        elif num[m] == numABuscar:
            print(cont)
            return m


def busquedaExhaustiva(num):
    cont = 0
    for i in range(len(num)):
        cont += 1
        if num[i] == 66:
            print(cont)
            return i


num = [1,3,5,6,7,8,9,44,66]
numABuscar = 66
print(busquedaBinaria(num))
print(busquedaExhaustiva(num))

File: busquedaBinaria/binaria.java
package busquedaBinaria;

public class binaria {

    private static int busqBinaria (int[] num) {
        int izquierda = 0;
        int derecha = num.length -1;
        int numABuscar = 66;
        int medio = 0;

        while(izquierda <= derecha){
            medio = (izquierda+derecha/2);

            if(num[medio] < numABuscar){
                izquierda = medio +1;
            }
            else if(num[medio] > numABuscar){
                derecha = medio - 1;
            }
            else if(num[medio] == numABuscar){
                break;
            }    
        }
        return medio;
    }

       
    public static void main(String[] args) {
        int num[] = {1, 3, 4, 5, 9, 48, 52, 66};
        busqBinaria(num);
        System.out.println(busqBinaria(num));
    }

}
